pypeit/scripts/chk_calibs.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+#
+# See top-level LICENSE file for Copyright information
+#
+# -*- coding: utf-8 -*-
+"""
+This script examines a set of files and indicates which do and
+which do not have sufficient calibs
+"""
+import argparse
+from pypeit import defs
+
+from IPython import embed
+
+def parser(options=None):
+    # TODO: Add argument that specifies the log file
+    parser = argparse.ArgumentParser(description="Script to setup a PypeIt run [v3]")
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('-r', '--root', type=str, default=None,
+                       help='File path+root, e.g. /data/Kast/b ')
+    parser.add_argument('-s', '--spectrograph', default=None, type=str,
+                        help='A valid spectrograph identifier: {0}'.format(
+                                ', '.join(defs.pypeit_spectrographs)))
+    parser.add_argument('-e', '--extension', default='.fits',
+                        help='File extension; compression indicators (e.g. .gz) not required.')
+
+    return parser.parse_args() if options is None else parser.parse_args(options)
+
+
+def main(args):
+    """
+
+    Args:
+        args:
+
+    Returns:
+        astropy.table.Table:
+
+    """
+
+    import os
+    import numpy as np
+
+    from astropy import table
+
+    from pypeit.pypeitsetup import PypeItSetup
+    from pypeit import calibrations
+    from pypeit import msgs
+    from pypeit.par import PypeItPar
+
+    # Check that the spectrograph is provided if using a file root
+    if args.root is not None:
+        if args.spectrograph is None:
+            raise ValueError('Must provide spectrograph identifier with file root.')
+        # Check that input spectrograph is supported
+        instruments_served = defs.pypeit_spectrographs
+        if args.spectrograph not in instruments_served:
+            raise ValueError('Instrument \'{0}\' unknown to PypeIt.\n'.format(args.spectrograph)
+                             + '\tOptions are: {0}\n'.format(', '.join(instruments_served))
+                             + '\tSelect an available instrument or consult the documentation '
+                             + 'on how to add a new instrument.')
+
+    # Initialize PypeItSetup based on the arguments
+    if args.root is not None:
+        ps = PypeItSetup.from_file_root(args.root, args.spectrograph, extension=args.extension)
+    else:
+        # Should never reach here
+        raise IOError('Need to set -r !!')
```

Comment:
Don't need this if you use argparse to make root a required argument.

Reply:
removed